### ISSUE TITLE
The first page is blank when building a project with the MNRAS LaTeX class 

### ISSUE THREAD

**Issue by samvaughan** [showyourwork/showyourwork#253]
Hi, 

Love the project! I'm applying it to a paper I'm currently working on which will be submitted to Monthly Notices. When using showyourwork to build the final manuscript, the final ms.pdf has a blank first page (apart from the head, page numbers and showyourwork stamp). Everything else after that appears to be working fine. 

I made a minimal example in the repo [here](https://github.com/samvaughan/test-mnras), and I've uploaded the logs from my .showyourwork folder and the final PDF which gets created so you can see what I mean:

[tectonic.log](https://github.com/showyourwork/showyourwork/files/10430213/tectonic.log)
[snakemake.log](https://github.com/showyourwork/showyourwork/files/10430214/snakemake.log)
[showyourwork.log](https://github.com/showyourwork/showyourwork/files/10430215/showyourwork.log)
[ms.pdf](https://github.com/showyourwork/showyourwork/files/10430199/ms.pdf)

Thanks very much for your help!

**Comment by dfm** [comment]
Very strange! What happens if you disable the stamp? You should be able [do that](https://show-your.work/en/latest/config/#stamp-enabled) by including:

```yaml
stamp:
  enabled: false
```

in your `showyourwork.yml`. I remember that there were some issues with the stamp messing up page layout when first added...

**Comment by samvaughan** [comment]
Same result unfortunately- still a blank first page...

**Comment by dfm** [comment]
That suggests to me that it might a tectonic issue rather than showyourwork, but we'd need to dig some more because there could certainly be other sources of trouble!

**Comment by samvaughan** [comment]
Yeah okay, that makes sense. I don't know enough about Latex to see anything obviously wrong in the tectonic log but I'll try and see if I get the same result using tectonic on its own rather than calling showyourwork

**Comment by samvaughan** [comment]
Running tectonic on the ms.tex file on its own doesn't give a blank page at the start- everything looks normal. 

In my directory,  showyourwork has installed its conda environment in ```.snakemake/conda/f24dd90b500379b1260c5449bebfb4bd_/```. I ran the following:

```bash
.snakemake/conda/f24dd90b500379b1260c5449bebfb4bd_/bin/tectonic src/tex/ms.tex 
```
and this gave the [following](https://github.com/showyourwork/showyourwork/files/10463243/running_tectonic_directly.log)  long ```tectonic``` output before finishing normally.

So if tectonic makes the correct PDF file, do you know what might be interfering with the output that showyourwork gives? Does showyourwork pass any extra arguments to tectonic maybe?

**Comment by dfm** [comment]
Strange! Thanks for the update. I'll take a closer look at this ASAP (not before later this week - sorry!).

**Comment by dfm** [comment]
I've narrowed it down to identify that the issue comes from this package (`showkeys`) that is automatically included when building with `showyourwork`:

https://github.com/showyourwork/showyourwork/blob/02a746eb844ee4fb054cc2cb1e4b1ef37c1bcff4/showyourwork/workflow/resources/styles/build.tex#L11

I'm not exactly sure what the issue is yet, but this at least gives us an idea of where to dig!

**Comment by dfm** [comment]
Sorry about the slow follow up on this, but I figured out where this issue was coming from! (Thanks to @katiebreivik: https://github.com/katiebreivik/cataclysmic-pileup-paper/pull/1)

At the top of the MNRAS template `.tex` file, you'll find the following lines:

```tex
% Don't change these lines
\begin{document}
\label{firstpage}
\pagerange{\pageref{firstpage}--\pageref{lastpage}}
```

And the `\label{firstpage}` line is the thing causing the blank page. If you ignore the comment and replace that with:

```diff
% Don't change these lines
\begin{document}
- \label{firstpage}
- \pagerange{\pageref{firstpage}--\pageref{lastpage}}
+ \pagerange{?--?}
```

Everything seems to work! Although you may also want to remove the `\label{lastpage}` at the end of the document.

I expect this is something we can fix in SYW so I want to keep this issue open for now.

**Comment by samvaughan** [comment]
`% Don't change these lines` 😄 

Awesome! Looks like that's fixed it. Thanks!

**Comment by HealthyPear** [comment]
I tested this as of today (`showyourwork 0.4.4.dev183+g297242b` and `mnras_template.tex v3.2`) and this is still happening.

Though I do not think it is a bug: I have a suspicion that those two lines in conjunction with `\maketitle` are needed by MNRAS to generate the correct page range in the journal header and metadata.

`\maketitle` triggers some page-breaking logic that assumes you will define both `\label{firstpage}` and `\pagerange{...}`.

If `\pagerange{...}` contains unresolved references (e.g. using` \pageref{firstpage}` before the label is placed), the class may get confused during the first LaTeX pass, leading to this blank output.

So the workaround described here solves the "problem" by providing a placeholder that doesn't depend on unresolved references, thus avoiding the blank page.

I'm curious @samvaughan did you happen to submit the paper in the end? Did they ask you to revert this change eventually?

**Comment by HealthyPear** [comment]
By the way I created a template repository that people can use out-of-the-box

https://github.com/HealthyPear/mnras-showyourwork-paper-template

**Comment by samvaughan** [comment]
@HealthyPear I didn't end up submitting the show-your-work version to the journal in the end- I can't remember what stopped me though I'm afraid. Great to see you've made a template that will work out of the box!

**Comment by mkenworthy** [comment]
Just dropping in here in 2026 to say **thank you** for the template and for the debugging of this issue!
